wrapanapi/systems/virtualcenter.py

```diff
@@ -162,6 +162,37 @@ def host(self):
         self.refresh()
         return self.raw.runtime.host.name
 
+    @staticmethod
+    def _progress_log_callback(logger, source, destination, progress):
+        logger.info("Provisioning progress {}->{}: {}".format(
+            source, destination, str(progress)))
+
+    @staticmethod
+    def _get_loc_of_vm(source_template, progress_callback):
+        """ Get the location where the inventory object will be stored"""
+        folder = getattr(source_template.parent.parent, "vmParent", None) or source_template.parent
+        progress_callback("Picked folder `{}`".format(folder.name))
+        return folder
+
+    @staticmethod
+    def _set_vm_clone_spec(power_on, template, cpu, ram, vm_reloc_spec):
+        """Set properties for Virtual Machine Cloning Operation specification """
+
+        vm_clone_spec = vim.VirtualMachineCloneSpec()
+
+        vm_clone_spec.powerOn = power_on
+        vm_clone_spec.template = template
+        vm_clone_spec.location = vm_reloc_spec
+        vm_clone_spec.snapshot = None
+
+        # Optional arguments
+        if cpu is not None:
```

Comment:
if those arguments are optional, shouldn't those be optional in method spec and be put to the end?